### ISSUE TITLE
`pdf()` method returns inverse density instead of density

### ISSUE THREAD

**Issue by ihagverdi** [PriorLabs/TabPFN#796]
### Describe the bug

The `pdf()` method in `FullSupportBarDistribution` claims to return the probability density function, but actually returns the **inverse** of the density (1/p(y)) instead of p(y).

Location:
File: `tabpfn/architectures/base/bar_distribution.py`
Class: `FullSupportBarDistribution`
Method: `pdf()`

### Steps/Code to Reproduce

The function below is wrong:
def pdf(self, logits: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
    """Probability density function at y."""
    return torch.exp(self.forward(logits, y))

### Expected Results

Returns the probability density at y.

### Actual Results

Returns the inverse probability density at y.

### Versions

```shell
PyTorch version: 2.10.0+cu130
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro (10.0.26200 64-bit)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.13.11 (main, Jan 14 2026, 19:29:23) [MSC v.1944 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26200-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to:
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4050 Laptop GPU
Nvidia driver version: 581.86
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Name: 13th Gen Intel(R) Core(TM) i5-13450HX
Manufacturer: GenuineIntel
Family: 205
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 2400
MaxClockSpeed: 2400
L2CacheSize: 9728
L2CacheSpeed: None
Revision: None

Dependency Versions:
--------------------
tabpfn: 6.4.1
torch: 2.10.0+cu130
numpy: 2.3.5
scipy: 1.17.0
pandas: 2.3.3
scikit-learn: 1.8.0
typing_extensions: 4.15.0
einops: 0.8.2
huggingface-hub: 1.4.1
```

**Comment by klemens-floege** [comment]
Dear @ihagverdi, thanks you so much for raising the issue. I was able to reproduce the bug. I think the fix could be as simple as changing the sign in the `pdf()` function: 

```
 def pdf(self, logits: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
        """Probability density function at y."""
        return torch.exp(-self.forward(logits, y))
```

but I will get back to you.

**Comment by klemens-floege** [comment]
I made a fix https://github.com/PriorLabs/TabPFN/pull/799, turns out we do not use the `pdf()` function inside of our repo so I think there are no further ramifications; anyway thanks a lot of raising the issue and we will fix the pdf function. Sorry if this caused any inconvenience.

**Comment by noahho** [comment]
@klemens-floege thanks for fixing! We use the pdf in extensions (https://github.com/PriorLabs/tabpfn-extensions/blob/cce79c61a8b8434be6eb9a78e7d30d446384d900/src/tabpfn_extensions/unsupervised/unsupervised.py#L641), so we should fix these places now, if we can't fix these today, maybe revert this change for now as these will be broken for now afaik?
